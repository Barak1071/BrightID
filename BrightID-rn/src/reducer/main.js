// @flow

import {
  USER_TRUST_SCORE,
  GROUPS_COUNT,
  SEARCH_PARAM,
  UPDATE_CONNECTIONS,
  ADD_CONNECTION,
  UPDATE_USER_DATA,
  REMOVE_USER_DATA,
  USER_AVATAR,
  REFRESH_NEARBY_PEOPLE,
  CONNECT_PUBLICKEY,
  CONNECT_NAMEORNYM,
  CONNECT_TIMESTAMP,
  CONNECT_TRUST_SCORE,
  CONNECT_AVATAR,
  ARBITER,
  RTC_ID,
  RESET_WEBRTC,
  BOX_KEYPAIR,
} from '../actions';

const arbiterSchema = {
  ALPHA: {
    OFFER: '',
    ICE_CANDIDATE: '',
  },
  ZETA: {
    ANSWER: '',
    ICE_CANDIDATE: '',
  },
};

/**
 * INITIAL STATE
 * structure the state of the app here
 *
 * @param trustScore String
 * @param name String
 * @param userAvatar Image
 * @param groupsCount Number
 * @param searchParam String
 * @param connections Array => Object
 */

export const initialState = {
  trustScore: '',
  nameornym: '',
  userAvatar: '',
  groupsCount: '',
  searchParam: '',
  connections: [
    {
      publicKey: new Uint8Array(),
      name: '',
      avatar: '',
      connectionDate: '',
      trustScore: '',
    },
  ],
  nearbyPeople: [],
  publicKey: '',
  secretKey: '',
  connectNameornym: '',
  connectPublicKey: '',
  connectTimestamp: '',
  connectTrustScore: '',
  connectAvatar: '',
  connectBoxKeypair: {
    publicKey: '',
    secretKey: '',
    nonce: 'wadata',
  },
  rtcId: '',
  arbiter: arbiterSchema,
};

export const mainReducer = (state = initialState, action) => {
  switch (action.type) {
    case USER_TRUST_SCORE:
      return {
        ...state,
        trustScore: action.trustScore,
      };
    case GROUPS_COUNT:
      return {
        ...state,
        groupsCount: action.groupsCount,
      };
    case USER_AVATAR:
      return {
        ...state,
        userAvatar: action.userAvatar,
      };
    case SEARCH_PARAM:
      return {
        ...state,
        searchParam: action.value,
      };
    case UPDATE_CONNECTIONS:
      return {
        ...state,
        connections: action.connections,
      };
    case ADD_CONNECTION:
      return {
        ...state,
        connections: [...state.connections, action.connection],
      };
    case UPDATE_USER_DATA:
      return {
        ...state,
        userAvatar:
          action.userAvatar || require('../static/default_avatar.jpg'),
        nameornym: action.nameornym,
        publicKey: action.publicKey,
        secretKey: action.secretKey,
      };
    case REMOVE_USER_DATA:
      return {
        ...state,
        userAvatar: '',
        nameornym: '',
        publicKey: '',
        secretKey: '',
      };

    case REFRESH_NEARBY_PEOPLE:
      return {
        ...state,
        nearbyPeople: action.nearbyPeople,
      };
    case CONNECT_PUBLICKEY:
      return {
        ...state,
        connectPublicKey: action.publicKey,
      };
    case CONNECT_NAMEORNYM:
      return {
        ...state,
        connectNameornym: action.nameornym,
      };
    case CONNECT_AVATAR:
      return {
        ...state,
        connectAvatar: action.avatar,
      };
    case CONNECT_TIMESTAMP:
      return {
        ...state,
        connectTimestamp: action.timestamp,
      };
    case CONNECT_TRUST_SCORE:
      return {
        ...state,
        connectTrustScore: action.trustScore,
      };
    case BOX_KEYPAIR:
      return {
        ...state,
        connectBoxKeypair: {
          ...state.connectBoxKeypair,
          publicKey: action.keypair.publicKey,
          secretKey: action.keypair.secretKey,
        },
      };
    case ARBITER:
      return {
        ...state,
        arbiter: action.arbiter,
      };
    case RTC_ID:
      return {
        ...state,
        rtcId: action.rtcId,
      };
    case RESET_WEBRTC:
      return {
        ...state,
        rtcId: '',
        arbiter: arbiterSchema,
        connectBoxKeypair: {
          ...state.connectBoxKeypair,
          publicKey: '',
          secretKey: '',
        },
      };
    default:
      return state;
  }
};

// unnecessary for now, but when the app gets larger, combine reducers here

export default mainReducer;
