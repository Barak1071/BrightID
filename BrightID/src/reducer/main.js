// @flow

import {
  USER_TRUST_SCORE,
  GROUPS_COUNT,
  SEARCH_PARAM,
  UPDATE_CONNECTIONS,
  CONNECTIONS_SORT,
  ADD_CONNECTION,
  UPDATE_USER_DATA,
  REMOVE_USER_DATA,
  USER_AVATAR,
  REFRESH_NEARBY_PEOPLE,
  CONNECT_PUBLICKEY,
  CONNECT_NAMEORNYM,
  CONNECT_TIMESTAMP,
  CONNECT_TRUST_SCORE,
  CONNECT_AVATAR,
  CONNECT_RECIEVED_TIMESTAMP,
  CONNECT_RECIEVED_NAMEORNYM,
  CONNECT_RECIEVED_PUBLICKEY,
  CONNECT_RECIEVED_AVATAR,
  CONNECT_RECIEVED_TRUSTSCORE,
  SET_CONNECT_QR_DATA,
  REMOVE_CONNECT_QR_DATA,
  REMOVE_CONNECTION,
  ARBITER,
  RTC_ID,
  RESET_WEBRTC,
  BOX_KEYPAIR,
  SET_PREVIEW,
  RESET_PREVIEW,
  SET_CONNECT_USER_DATA,
  REMOVE_CONNECT_USER_DATA,
  SET_ENCRYPTED_USER_DATA,
  REMOVE_ENCRYPTED_USER_DATA,
} from '../actions';

const arbiterSchema = {
  USERA: {
    OFFER: '',
    ICE_CANDIDATE: '',
  },
  USERB: {
    ANSWER: '',
    ICE_CANDIDATE: '',
  },
};

/**
 * INITIAL STATE
 * structure the state of the app here
 *
 * @param trustScore String
 * @param name String
 * @param userAvatar Image
 * @param groupsCount Number
 * @param searchParam String
 * @param connections Array => Object
 */

export const initialState = {
  trustScore: '',
  nameornym: '',
  userAvatar: '',
  groupsCount: '',
  searchParam: '',
  connections: [],
  nearbyPeople: [],
  publicKey: '',
  secretKey: '',
  connectionsSort: '',
  connectQrData: { aesKey: '', ipAddress: {}, uuid: '', user: '' },
  connectUserData: { publicKey: '', avatar: '', nameornym: '', timestamp: '' },
  encryptedUserData: '',
};

export const mainReducer = (state = initialState, action) => {
  const connections = state.connections.slice();
  switch (action.type) {
    case USER_TRUST_SCORE:
      return {
        ...state,
        trustScore: action.trustScore,
      };
    case GROUPS_COUNT:
      return {
        ...state,
        groupsCount: action.groupsCount,
      };
    case USER_AVATAR:
      return {
        ...state,
        userAvatar: action.userAvatar,
      };
    case SEARCH_PARAM:
      return {
        ...state,
        searchParam: action.value,
      };
    case UPDATE_CONNECTIONS:
      return {
        ...state,
        connections: action.connections,
      };
    case CONNECTIONS_SORT:
      return {
        ...state,
        connectionsSort: action.connectionsSort,
      };
    case ADD_CONNECTION:
      return {
        ...state,
        connections: [action.connection, ...connections],
      };
    case SET_ENCRYPTED_USER_DATA:
      return {
        ...state,
        encryptedUserData: action.encryptedData,
      };
    case REMOVE_ENCRYPTED_USER_DATA:
      return {
        ...state,
        encryptedUserData: '',
      };
    case REMOVE_CONNECTION:
      return {
        ...state,
        connections: connections.filter(
          (val) =>
            JSON.stringify(val.publicKey) !== JSON.stringify(action.publicKey),
        ),
      };
    case UPDATE_USER_DATA:
      return {
        ...state,
        userAvatar: action.userAvatar,
        nameornym: action.nameornym,
        publicKey: action.publicKey,
        secretKey: action.secretKey,
      };
    case REMOVE_USER_DATA:
      return {
        ...state,
        userAvatar: '',
        nameornym: '',
        publicKey: '',
        secretKey: '',
      };

    case REFRESH_NEARBY_PEOPLE:
      return {
        ...state,
        nearbyPeople: action.nearbyPeople,
      };
    case CONNECT_PUBLICKEY:
      return {
        ...state,
        connectPublicKey: action.publicKey,
      };
    case SET_CONNECT_QR_DATA:
      return {
        ...state,
        connectQrData: action.connectQrData,
      };
    case REMOVE_CONNECT_QR_DATA:
      return {
        ...state,
        connectQrData: { aesKey: '', ipAddress: {}, uuid: '', user: '' },
      };
    case SET_CONNECT_USER_DATA:
      return {
        ...state,
        connectUserData: action.connectUserData,
      };
    case REMOVE_CONNECT_USER_DATA:
      return {
        ...state,
        connectUserData: {
          publicKey: '',
          avatar: '',
          nameornym: '',
          timestamp: '',
        },
      };
    case CONNECT_NAMEORNYM:
      return {
        ...state,
        connectNameornym: action.nameornym,
      };
    case CONNECT_AVATAR:
      return {
        ...state,
        connectAvatar: action.avatar,
      };
    case CONNECT_TIMESTAMP:
      return {
        ...state,
        connectTimestamp: action.timestamp,
      };
    case CONNECT_TRUST_SCORE:
      return {
        ...state,
        connectTrustScore: action.trustScore,
      };
    case CONNECT_RECIEVED_NAMEORNYM:
      return {
        ...state,
        connectRecievedNameornym: true,
      };
    case CONNECT_RECIEVED_TRUSTSCORE:
      return {
        ...state,
        connectRecievedTrustScore: true,
      };
    case CONNECT_RECIEVED_PUBLICKEY:
      return {
        ...state,
        connectRecievedPublicKey: true,
      };
    case CONNECT_RECIEVED_TIMESTAMP:
      return {
        ...state,
        connectRecievedTimestamp: true,
      };
    case CONNECT_RECIEVED_AVATAR:
      return {
        ...state,
        connectRecievedAvatar: true,
      };
    case BOX_KEYPAIR:
      return {
        ...state,
        connectBoxKeypair: {
          ...state.connectBoxKeypair,
          publicKey: action.keypair.publicKey,
          secretKey: action.keypair.secretKey,
        },
      };
    case ARBITER:
      return {
        ...state,
        arbiter: action.arbiter,
      };
    case RTC_ID:
      return {
        ...state,
        rtcId: action.rtcId,
      };
    case RESET_WEBRTC:
      return {
        ...state,
        rtcId: '',
        arbiter: arbiterSchema,
        connectTimestamp: '',
        connectNameornym: '',
        connectTrustScore: '',
        connectAvatar: '',
        connectPublicKey: '',
        connectRecievedNameornym: false,
        connectRecievedPublicKey: false,
        connectRecievedTimestamp: false,
        connectRecievedTrustScore: false,
        connectRecievedAvatar: false,
        connectBoxKeypair: {
          ...state.connectBoxKeypair,
          publicKey: '',
          secretKey: '',
        },
      };
    case SET_PREVIEW:
      return {
        ...state,
        previewTimestamp: state.connectTimestamp,
        previewNameornym: state.connectNameornym,
        previewTrustScore: state.connectTrustScore,
        previewAvatar: state.connectAvatar,
        previewPublicKey: state.connectPublicKey,
      };
    case RESET_PREVIEW:
      return {
        ...state,
        previewTimestamp: '',
        previewNameornym: '',
        previewTrustScore: '',
        previewAvatar: '',
        previewPublicKey: '',
      };
    default:
      return state;
  }
};

// unnecessary for now, but when the app gets larger, combine reducers here

export default mainReducer;
