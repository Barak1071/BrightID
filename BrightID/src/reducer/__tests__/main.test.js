// @flow

import nacl from 'tweetnacl';
import { mainReducer, initialState } from '../main';
import {
  USER_TRUST_SCORE,
  GROUPS_COUNT,
  SEARCH_PARAM,
  UPDATE_CONNECTIONS,
  UPDATE_USER_DATA,
  REMOVE_USER_DATA,
} from '../../actions';

describe('main reducer', () => {
  test('should return the initial state', () => {
    expect(mainReducer(undefined, {})).toMatchSnapshot();
  });

  test('should update USER_TRUST_SCORE', () => {
    expect(
      mainReducer(undefined, {
        type: USER_TRUST_SCORE,
        trustScore: '99.9',
      }),
    ).toMatchSnapshot();
  });

  test('should update GROUPS_COUNT', () => {
    expect(
      mainReducer(undefined, {
        type: GROUPS_COUNT,
        groupsCount: 64,
      }),
    ).toMatchSnapshot();
  });

  test('should update SEARCH_PARAM', () => {
    expect(
      mainReducer(undefined, {
        type: SEARCH_PARAM,
        value: 'hi john',
      }),
    ).toMatchSnapshot();
  });

  test('should UPDATE_CONNECTIONS', () => {
    expect(
      mainReducer(undefined, {
        type: UPDATE_CONNECTIONS,
        connections: [
          {
            publicKey: [],
            name: 'Test User',
            avatar: 'todo...',
            connectionDate: 1532537998586,
            trustScore: '85.1',
          },
        ],
      }),
    ).toMatchSnapshot();
  });

  test('should UPDATE_USER_DATA', () => {
    expect(
      mainReducer(undefined, {
        type: UPDATE_USER_DATA,
        publicKey: [],
        secretKey: [],
        nameornym: 'Test User',
        userAvatar: 'todo...',
      }),
    ).toMatchSnapshot();
  });

  test('should REMOVE_USER_DATA', () => {
    expect(
      mainReducer(undefined, {
        type: REMOVE_USER_DATA,
        publicKey: new Uint8Array(32),
        secretKey: new Uint8Array(64),
        name: 'Test User',
        userAvatar: 'todo...',
      }),
    ).toMatchSnapshot();
  });
});

test('should update USER_TRUST_SCORE', () => {
  expect(
    mainReducer(undefined, {
      type: USER_TRUST_SCORE,
      trustScore: '99.9',
    }),
  ).toEqual({
    ...initialState,
    trustScore: '99.9',
  });
});

// test('should update GROUPS_COUNT', () => {
//   expect(
//     mainReducer(undefined, {
//       type: GROUPS_COUNT,
//       groupsCount: 64,
//     }),
//   ).toEqual({
//     ...initialState,
//     groupsCount: 64,
//   });
// });

// test('should update SEARCH_PARAM', () => {
//   expect(
//     mainReducer(undefined, {
//       type: SEARCH_PARAM,
//       value: 'hi john',
//     }),
//   ).toEqual({
//     ...initialState,
//     searchParam: 'hi john',
//   });
// });

// test('should UPDATE_CONNECTIONS', () => {
//   const pk = new Uint8Array(32);
//   expect(
//     mainReducer(undefined, {
//       type: UPDATE_CONNECTIONS,
//       connections: [
//         {
//           publicKey: pk,
//           name: 'Test User',
//           avatar: 'todo...',
//           connectionDate: 1532537998586,
//           trustScore: '85.1',
//         },
//       ],
//     }),
//   ).toEqual({
//     ...initialState,
//     connections: [
//       {
//         publicKey: pk,
//         name: 'Test User',
//         avatar: 'todo...',
//         connectionDate: 1532537998586,
//         trustScore: '85.1',
//       },
//     ],
//   });
// });

// test('should UPDATE_USER_DATA', () => {
//   const { publicKey, secretKey } = nacl.sign.keyPair();
//   expect(
//     mainReducer(undefined, {
//       type: UPDATE_USER_DATA,
//       publicKey,
//       secretKey,
//       nameornym: 'Test User',
//       userAvatar: 'todo...',
//     }),
//   ).toEqual({
//     ...initialState,
//     publicKey,
//     secretKey,
//     name: 'Test User',
//     userAvatar: 'todo...',
//   });
// });

// test('should REMOVE_USER_DATA', () => {
//   expect(
//     mainReducer(undefined, {
//       type: REMOVE_USER_DATA,
//       publicKey: new Uint8Array(32),
//       secretKey: new Uint8Array(64),
//       name: 'Test User',
//       userAvatar: 'todo...',
//     }),
//   ).toEqual({
//     ...initialState,
//     publicKey: '',
//     secretKey: '',
//     name: '',
//     userAvatar: '',
//   });
// });

// test('should set PUBLICKEY2', () => {
//   const { publicKey } = nacl.sign.keyPair();
//   expect(
//     mainReducer(undefined, {
//       type: PUBLICKEY2,
//       publicKey2: publicKey,
//     }),
//   ).toEqual({
//     ...initialState,
//     publicKey2: publicKey,
//   });
// });
