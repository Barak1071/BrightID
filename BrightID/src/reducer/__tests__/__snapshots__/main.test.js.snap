// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main reducer should REMOVE_USER_DATA 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "score": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "score": "",
  "photo": "",
}
`;

exports[`main reducer should UPDATE_CONNECTIONS 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "todo...",
      "connectionDate": 1532537998586,
      "name": "Test User",
      "publicKey": Array [],
      "score": "85.1",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "score": "",
  "photo": "",
}
`;

exports[`main reducer should UPDATE_USER_DATA 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "score": "",
    },
  ],
  "groupsCount": "",
  "name": "Test User",
  "nearbyPeople": Array [],
  "publicKey": Array [],
  "publicKey2": "",
  "searchParam": "",
  "secretKey": Array [],
  "score": "",
  "photo": "todo...",
}
`;

exports[`main reducer should return the initial state 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "score": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "score": "",
  "photo": "",
}
`;

exports[`main reducer should set PUBLICKEY2 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "score": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": Array [],
  "searchParam": "",
  "secretKey": "",
  "score": "",
  "photo": "",
}
`;

exports[`main reducer should update GROUPS_COUNT 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "score": "",
    },
  ],
  "groupsCount": 64,
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "score": "",
  "photo": "",
}
`;

exports[`main reducer should update SEARCH_PARAM 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "score": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "hi john",
  "secretKey": "",
  "score": "",
  "photo": "",
}
`;

exports[`main reducer should update USER_SCORE 1`] = `
Object {
  "connections": Array [
    Object {
      "photo": "",
      "connectionDate": "",
      "name": "",
      "publicKey": Array [],
      "score": "",
    },
  ],
  "groupsCount": "",
  "name": "",
  "nearbyPeople": Array [],
  "publicKey": "",
  "publicKey2": "",
  "searchParam": "",
  "secretKey": "",
  "score": "99.9",
  "photo": "",
}
`;
